diy_array_list/src/main/java/DenArrayList.java

```diff
@@ -0,0 +1,203 @@
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.List;
+import java.util.ListIterator;
+
+public class DenArrayList<T> implements List<T> {
+
+    private int sizeArr = 0;
+
+    T[] arr;
```

Comment:
```suggestion
    private T[] arr;
```
Поля лучше всегда `private`, а где можно там `private final`.
Чтобы снаружи никто не испортил состояние объекта.